op-bindings/bindgen/main.go

```diff
@@ -0,0 +1,140 @@
+package main
+
+import (
+	"fmt"
+	"os"
+
+	oplog "github.com/ethereum-optimism/optimism/op-service/log"
+	gethLog "github.com/ethereum/go-ethereum/log"
+	"github.com/urfave/cli/v2"
+)
+
+type bindGenGeneratorBase struct {
+	metadataOut         string
+	bindingsPackageName string
+	monorepoBasePath    string
+	contractsListPath   string
+	logger              gethLog.Logger
+}
+
+const (
+	// Base Flags
+	MetadataOutFlagName         = "metadata-out"
+	BindingsPackageNameFlagName = "bindings-package"
+	MonoRepoBaseFlagName        = "monorepo-base"
+	ContractsListFlagName       = "contracts-list"
+	LogLevelFlagName            = "log.level"
+
+	// Local Contracts Flags
+	SourceMapsListFlagName = "source-maps-list"
+	ForgeArtifactsFlagName = "forge-artifacts"
+)
+
+func main() {
+	oplog.SetupDefaults()
+
+	app := &cli.App{
+		Name:  "BindGen",
+		Usage: "Generate contract bindings using Foundry artifacts and/or remotely sourced contract data",
+		Commands: []*cli.Command{
+			{
+				Name:  "generate",
+				Usage: "Generate contract bindings",
+				Flags: baseFlags(),
+				Subcommands: []*cli.Command{
+					{
+						Name:   "local",
+						Usage:  "Generate bindings for locally sourced contracts",
+						Flags:  localFlags(),
+						Action: generateBindings,
+					},
+				},
+			},
+		},
+	}
+
+	if err := app.Run(os.Args); err != nil {
+		gethLog.Crit("Error staring CLI app", "error", err.Error())
+	}
+}
+
+func setupLogger(c *cli.Context) (gethLog.Logger, error) {
+	logger := oplog.NewLogger(oplog.AppOut(c), oplog.ReadCLIConfig(c))
+	oplog.SetGlobalLogHandler(logger.GetHandler())
+	return logger, nil
+}
+
+func generateBindings(c *cli.Context) error {
+	logger, _ := setupLogger(c)
+
+	switch c.Command.Name {
+	case "local":
+		localBindingsGenerator := parseConfigLocal(logger, c)
+		if err := localBindingsGenerator.generateBindings(); err != nil {
+			gethLog.Crit("Error generating local bindings", "error", err.Error())
+		}
+		return nil
```

Comment:
No need to call `Crit` anywhere else since `app.Run` will forward any returned errors into crit
```suggestion
		if err := localBindingsGenerator.generateBindings(); err != nil {
			return fmt.Errorf("Error generating local bindings: %w", err)
		}
		return nil
```